src/stringParamAllocator.js

```diff
@@ -0,0 +1,92 @@
+import { isArrayAllStrings } from './validation.js';
+
+/* global Module */
+export default class ParamParser {
+    constructor(args, errorHandling) {
+        let self = this;
+
+        if (!isArrayAllStrings(args)) {
+            throw new Error('All items in the argument list must be strings');
+        }
+        // So first, we need to allocate Emscripten heap space sufficient to store each string as a
+        // null-terminated C string.
+        // Because the C function signature is char **, this array of pointers is going to need to
+        // get copied into Emscripten heap space eventually, so we're going to prepare by storing
+        // the pointers as a typed array so that we can more easily copy it into heap space later.
+        let argPtrsArray = Uint32Array.from(
```

Comment:
Explicitly calling ```allocate``` and ```deallocate``` makes sense. We should explicitly call the parser to assign those values in memory.

For deallocation, what did you have in mind? I certainly would think the ```cleanup()``` logic would be included in this, but is anything missing?

Reply:
Yes, I think that's exactly right -- `allocate()` would perform most of the work that's happening in the constructor now, and `deallocate()` would handle clearing out the pointers allocated by `allocate()`. This would definitely be similar to the existing `cleanup()`, but the way I'm envisioning it would omit cleaning up the `usageErrPtr` or the `*OptionsPtr`, because these are not allocated inside the class.